### ISSUE TITLE
[v3] Implement new Concept Exercise: maps

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/fsharp#866]
This issue describes how to implement the `maps` concept exercise for the F# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the basics of the Concept of Maps in F#.

## Learning objectives

- Know of the existence of the `Map` type.
- Know how to define a map.
- Know that maps have structural equality.
- Know that maps are immutable.
- Know how to add and update elements in a map.
- Know how to find or access elements in a map by key.
- Know some basic map functions (like checking if a map is empty).

## Out of scope

- Memory and performance characteristics.

## Concepts

The Concepts this exercise unlocks are:

- `maps`: know of the existence of the `Map` type; know how to define a map; know that maps have structural equality; know that maps are immutable; know how to add and update elements in a map; know how to find or access elements in a map by key; know some basic map functions (like checking if a map is empty).

## Prerequisites

This exercise's prerequisites Concepts are:

- `higher-order-functions`: know how to use higher-order functions.
- `generic-types`: know how generic types work.

Any data types used in this exercise (e.g. `strings`) should also be added as prerequisites.

### Hints

- [Maps][maps]: introduction to maps.
- [Map module][map-module]: the `Map` module's functions.

### After

- [Maps][maps]: introduction to maps.
- [Map module][map-module]: the `Map` module's functions.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/fsharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[how-to-implement-a-concept-exercise]: https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/fsharp/exercises/concept/README.md#implemented-exercises
[reference]: https://github.com/exercism/v3/blob/master/reference/types/map.md
[analyzer]: https://github.com/exercism/fsharp-analyzer
[representer]: https://github.com/exercism/fsharp-representer
[meta-design]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/design.md
[meta-config.json]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/config.json
[docs-concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[maps]: https://www.tutorialspoint.com/fsharp/fsharp_maps.htm
[map-module]: https://en.wikibooks.org/wiki/F_Sharp_Programming/Sets_and_Maps#The_Map_Module
